### PR TITLE
[dygraph qat]  Use layer to calculate output scale 

### PR DESCRIPTION
**Pull request by juncaipeng**
### PR types

Others
### PR changes

Others
### Describe

Use layer to calculate output scale

### CODE REVIEW

File: python/paddle/fluid/contrib/slim/tests/test_imperative_out_scale.py [hunk 1]
```diff
@@ -500,7 +500,7 @@ def test_warning(self):
                           "generated inference model would not contain the " \
                           "out_threshold."
         num = get_vaild_warning_num(warning_message, w)
-        assert num == 1
+        assert num == 0
```
Comment (gfwm2013) [inline python/paddle/fluid/contrib/slim/tests/test_imperative_out_scale.py]: 建议删除该检查warning的单测

**Review by gfwm2013 — APPROVED** [review]

File: python/paddle/fluid/contrib/slim/tests/test_imperative_out_scale.py (hunk 1 above)
Comment (juncaipeng) [reply python/paddle/fluid/contrib/slim/tests/test_imperative_out_scale.py]: Done

**Review by luotao1 — APPROVED** [review]
LGTM

**Review by zhiqiu — APPROVED** [review]
LGTM for op_function_generator.cc

**Pull request merged**